src/SelfScreener/UnderlyingConditions.tsx

```diff
@@ -0,0 +1,140 @@
+import { useNavigation } from "@react-navigation/native"
+import { ScrollView } from "react-native"
+import React, { FunctionComponent } from "react"
+import { useTranslation } from "react-i18next"
+import { SelfScreenerStackScreens } from "../navigation"
+import { useSelfScreenerContext } from "../SelfScreenerContext"
+import { Button, GlobalText } from "../components"
+import { UnderlyingCondition } from "./selfScreener"
+import SymptomCheckbox from "./SymptomCheckbox"
+
+const UnderlyingConditions: FunctionComponent = () => {
+  const { t } = useTranslation()
+  const navigation = useNavigation()
+  const {
+    LUNG_DISEASE,
+    HEART_CONDITION,
+    WEAKENED_IMMUNE_SYSTEM,
+    OBESITY,
+    KIDNEY_DISEASE,
+    DIABETES,
+    LIVER_DISEASE,
+    HIGH_BLOOD_PRESSURE,
+    BLOOD_DISORDER,
+    CEREBROVASCULAR_DISEASE,
+    SMOKING,
+    PREGNANCY,
+  } = UnderlyingCondition
+  const {
+    underlyingConditions,
+    updateUnderlyingConditions,
+  } = useSelfScreenerContext()
+
+  const underlyingConditionToString = (condition: UnderlyingCondition) => {
+    switch (condition) {
+      case LUNG_DISEASE:
+        return t("self_screener.underlying_conditions.lung_disease")
+      case HEART_CONDITION:
+        return t("self_screener.underlying_conditions.heart_condition")
+      case WEAKENED_IMMUNE_SYSTEM:
+        return t("self_screener.underlying_conditions.weakened_immune_system")
+      case OBESITY:
+        return t("self_screener.underlying_conditions.obesity")
+      case KIDNEY_DISEASE:
+        return t("self_screener.underlying_conditions.kidney_disease")
+      case DIABETES:
+        return t("self_screener.underlying_conditions.diabetes")
+      case LIVER_DISEASE:
+        return t("self_screener.underlying_conditions.liver_disease")
+      case HIGH_BLOOD_PRESSURE:
+        return t("self_screener.underlying_conditions.high_blood_pressure")
+      case BLOOD_DISORDER:
+        return t("self_screener.underlying_conditions.blood_disorder")
+      case CEREBROVASCULAR_DISEASE:
+        return t("self_screener.underlying_conditions.cerebrovascular_disease")
+      case SMOKING:
+        return t("self_screener.underlying_conditions.smoking")
+      case PREGNANCY:
+        return t("self_screener.underlying_conditions.pregnancy")
+    }
+  }
+
+  const handleOnPressNext = () => {
+    navigation.navigate(SelfScreenerStackScreens.AgeRange)
+  }
+
+  return (
+    <ScrollView>
+      <GlobalText>
+        {t("self_screener.underlying_conditions.select_all_or_none")}
+      </GlobalText>
+      <SymptomCheckbox
```

Comment:
I'm 💯  on board with this but since we already created the case statement for the translations could we benefit from having a collection of the shape:
```
const symptomsAndTranslations: [UnderlyingCondition, string][] = [...]
```
And then iterating over it and displaying a `SymptomCheckBox` don't think it is a huge improvement but I also think it wouldn't make the component harder to read but maybe it will make it a bit shorter?

Reply:
Clever boi 👍